spring-boot-starter/autoconfigure/src/main/java/tech/cassandre/trading/bot/ta4j/BarContext.java

```diff
@@ -70,27 +80,73 @@
         this.startTime = newStartTime;
         this.endTime = startTime.plus(duration);
         this.close = newClose != null ? newClose.doubleValue() : 0;
-
-        this.low = newLow != null ? newLow.doubleValue() : close;
-        this.high = newHigh != null ? newHigh.doubleValue() : close;
         this.open = newOpen != null ? newOpen.doubleValue() : close;
-        this.volume = newVolume != null ? newVolume.doubleValue() : 0;
+
+        this.low = Math.min(close, open);
+        this.high = Math.max(close, open);
+
+
+        calculateInitialVolume(newVolume);
+    }
+
+    private void calculateInitialVolume(final Number newVolume) {
+        if (newVolume == null) {
+            return;
+        }
+        if (isDurationMoreThanDay()) {
+            this.volume = newVolume.doubleValue();
+        } else {
+            this.initialDayVolume = newVolume.doubleValue();
+        }
     }
 
     public boolean isAfter(final ZonedDateTime timestamp) {
         return timestamp.isAfter(endTime.minus(Duration.ofSeconds(1)));
     }
 
+    /**
+     * The contract of the update call is that it is called without timestamp
+     * and is always within bounds of one bar duration.
+     *
+     * @param newClose  new close - mandatory
+     * @param newVolume new volume to be updated (might be null)
+     */
     @SuppressWarnings("checkstyle:AvoidInlineConditionals")
-    public void update(final Number newLow, final Number newHigh, final Number newClose, final Number newVolume) {
+    public void update(final Number newClose, final Number newVolume) {
         if (newClose == null) {
             throw new IllegalArgumentException("Cannot update bar context without at least specifying close price");
         }
         close = newClose.doubleValue();
-        low = Math.min(low, newLow == null ? close : newLow.doubleValue());
-        high = Math.max(high, newHigh == null ? close : newHigh.doubleValue());
+        low = Math.min(low, close);
+        high = Math.max(high, close);
+
+        calculateVolume(newVolume);
+        updatesReceived++;
+    }
+
+    private void calculateVolume(final Number newVolume) {
+        if (newVolume == null) {
+            return;
+        }
+        if (!isDurationMoreThanDay() && initialDayVolume >= 0) {
+            volume = newVolume.doubleValue() - initialDayVolume;
```

Comment:
What is the startpoint of the current/new volume?
If its 24h, e.g. a different start point as the initialDayVolume, then we cant do the computation as shown here.
We would loose volume.

Example:
initialDayVolume: 100 Units 16:00 UTC (Traded in the last 24h)
newVolume: 90 Units 17:00 UTC (Traded in the last 24h)

This example means, that from 16:00 until 17:00 in the LAST day 10+X units have been traded.
And it means, that in the current day between 16 and 17:00 X have been traded.

So the correct result would be X, but we get -10.

So far I have no glue about how to solve this... have to think a bit about it.

(If the start point of initialDayVolume and currentVolume is the same, then we dont have problems)